jupyterlab/templates/logs_analyzer/analyze_prometheus_logs.py

```diff
@@ -0,0 +1,141 @@
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+class AnalyzePrometheusLogs:
+    """
+    This class analyze prometheus logs
+    """
+    TIMEOUT = 30
+    SLEEP = 3
+    CHUNK_SIZE = 8192
+
+    def __init__(self, s3_logs_url: str):
+        self.__s3_logs_url = s3_logs_url
+        self.__logs_dir = os.path.join(os.path.join(os.getcwd(), 'logs'))
+        self.__filename = self.__s3_logs_url.split('/')[-1]
+        self.__log_dir_path = os.path.join(self.__logs_dir, self.__filename)
+
+    def cleanup(self):
+        """
+        This method cleans up existing logs and Prometheus images
+        @return:
+        """
+        # Delete Prometheus container
+        os.system('podman rmi -f docker.io/prom/prometheus;')
+        # delete logs dir if exist
+        if os.path.exists(self.__logs_dir):
+            os.system(f"rm -rf {self.__logs_dir}")
+
+    def download_s3_logs(self, username: str, password: str):
+        """
+        This method downloads s3 logs
+        @param username:
+        @param password:
+        @return:
+        """
+        if not os.path.exists(self.__logs_dir):
+            os.mkdir(self.__logs_dir)
+
+        # create a session with the credentials
+        session = requests.Session()
+        session.auth = (username, password)
+
+        # download with progress bar
+        response = session.get(self.__s3_logs_url, stream=True)
+        size = int(response.headers.get('Content-Length', 0))
+
+        progress = widgets.IntProgress(description='Downloading', min=0, max=size)
+        display(progress)
+
+        with open(f"{self.__logs_dir}/{self.__filename}", 'wb') as f:
+            with tqdm.wrapattr(f, "write", total=size) as fileobj:
+                for chunk in response.iter_content(chunk_size=self.CHUNK_SIZE):
+                    if chunk:
+                        fileobj.write(chunk)
+                        progress.value += len(chunk)
+
+        logger.info('Download complete!')
+
+    def untar_and_chmod_prometheus_logs(self):
+        """
+        This method untars and sets the chmod for Prometheus logs, and returns the path to the 'promdb' directory
+        @return:
+        """
+        logger.info(f'untar download file {self.__log_dir_path}')
+        os.system(f"tar -xvf {self.__log_dir_path} -C {self.__logs_dir}")
+
+        promdb_file = [f for f in os.listdir(self.__log_dir_path.split('.')[0]) if f.startswith('promdb')]
+
+        logger.info(f"untar prometheus file: {os.path.join(self.__logs_dir, self.__filename.split('.')[0], promdb_file[0])}")
+        os.system(f"tar -xvf {os.path.join(self.__logs_dir, self.__filename.split('.')[0], promdb_file[0])} -C {os.path.join(self.__logs_dir, self.__filename.split('.')[0])}")
+
+        promdb_file = [f for f in os.listdir(self.__log_dir_path.split('.')[0]) if f.startswith('promdb') and not f.endswith('tar')]
+        promdb_dir_path = f"{os.path.join(self.__logs_dir, self.__filename.split('.')[0], promdb_file[0])}"
```

Comment:
Zeros are repeated. Can we make it as const?

Reply:
Yes